src/app/modules/gb-data-selection-module/accordion-components/gb-selection/gb-selection.component.ts

```diff
@@ -69,6 +72,62 @@ export class GbSelectionComponent implements OnInit {
     this.queryService.updateCounts_1();
   }
 
+  importCriteria() {
+    let uploadElm = document.getElementById('step1CriteriaFileUpload');
+    if (this.isUploadListenerNotAdded) {
+      uploadElm
+        .addEventListener('change', this.criteriaFileUpload.bind(this), false);
+      this.isUploadListenerNotAdded = false;
+      // reset the input path so that it will take the same file again
+      document.getElementById('step1CriteriaFileUpload')['value'] = '';
+    }
+    uploadElm.click();
+  }
+
+  criteriaFileUpload(event) {
+    let reader = new FileReader();
+    let file = event.target.files[0];
+    reader.onload = (function (e) {
+      let data = e.target['result'];
+      let query = this.parseFile(file, data);
+      this.queryService.restoreQuery(query);
+    }).bind(this);
+    reader.readAsText(file);
+  }
+
+  private parseFile(file: File, data: any) {
+    if (file.type === 'text/plain' ||
+      file.type === 'text/tab-separated-values' ||
+      file.type === 'text/csv' ||
+      file.type === '') {
+      // we assume the text contains a list of subject Ids
+      let query = {
+        'name': file.name.substr(0, file.name.indexOf('.')),
+        'patientsQuery': {
+          'type': 'patient_set',
+          'subjectIds': data.split('\n')
+        },
+        'observationsQuery': {}
+      };
+      return query;
+    } else if (file.type === 'application/json') {
+      let _json = JSON.parse(data);
+      // If the json is of standard format
+      if (_json['patientsQuery'] || _json['observationsQuery']) {
```

Comment:
we probably need to limit ourselves about which json format to accept in importing criteria into step 1: query import is capable of importing both patient and observation criteria, step 1 is only capable of importing patient query, and step 2 is only capably of importing observation query.

Reply:
e.g. for step 1 import, we only accept this json format: 
`"patientsQuery": {
    "type": "true"
  }`

Reply:
for step 2 import, we only parse the content in this json format:
`"observationsQuery": {
    "data": []
  }`

or 
`"names": []`

or 
`"paths": []`

Reply:
but I guess "observationsQuery" can still be a part of a file imported in step1, just it will be skipped. 
For example, if someone exported a query and want to import it to step 1, it would be possible without editing. The only part that would be used from the export file during the import would be "patientsQuery" part. 
What do you think, @beaugogh ?

Reply:
yes, exactly!